### ISSUE TITLE
Allow the StreamBuilder.AsYaml to have options to disable scrubbing

### ISSUE THREAD

**Issue by calmera** [benthosdev/benthos#1939]
TL;DR - Scrubbing control is broken as `Spec.SanitiseYAML` does not take the `SanitiseConfig` into account for scrubbing

When you want to output the result of your StreamBuilder, you can call the `AsYaml` function to return a scrubbed version of your configuration. This is great because it formats the configuration file in a more readable way than what we would get doing the yaml Marshalling ourselves. However, there does not seem to be a way to disable the scrubbing.

Underneath a dedicated scrubber is used (defined on config field level) to scrub the actual value in case the field contains a secret. A call to `AsYaml` will create a `SanitiseConfig` containing various options, one being to control whether scrubbing should happen or not. However, this config is not taken into account by the scrubber defined on the field.

Approach:
- Guard the scrubbing behavior in the `Spec.SanitiseYAML` based on `SanitiseConfig.ScrubSecrets`
- Make sure the `NewSanitiseConfig` is enabling scrubbing by default
- Allow GenerationOptions to be passed to the `StreamBuilder.AsYaml()` call
- Create a `WithScrubbingDisabled`option to be passed to the `StreamBuilder.AsYaml()` call
